src/components/CatalogLayersLegendItems.vue

```diff
@@ -91,22 +89,31 @@ export default {
       return catalogLayer && catalogLayer.getLegendUrl(params, { format: 'image/png', categories: layer.categories });
     },
 
-    async getLegendSrc(_layers) {
+    /**
+     * get legend src for visible layers
+     * @returns {undefined}
+     */
```

Comment:
Double check, I think the correct return would instead:

```suggestion
    /**
     * get legend src for visible layers
     * @returns {Promise<void>}
     */
```
it also seems that we are missing a `return` statement, otherwise how can the line following be executed?

```js
await this.getLegendSrc(this.layers); // it waits indefinitely?

changeLayersLegend.forEach(layer => layer.legend.change = false); 
```

**PS** `getLegendSrc()` perhaps you should think of a more descriptive name for this function if it returns no value (ref: `@returns {undefined}`)